app/controllers/autosign_controller.rb

```diff
@@ -17,9 +20,9 @@ def create
     setup_proxy
 
     if @api.set_autosign(params[:id])
-      process_success({:success_redirect => smart_proxy_autosign_index_path(@proxy), :object_name => 'puppet autosign entry'})
+      process_success({:success_redirect => smart_proxy_path(@proxy, :anchor => 'autosign'), :object_name => 'puppet autosign entry'})
```

Comment:
`create` and `delete` are the same, please consider extracting the common behavior to a shared method.